### DISCUSSION TITLE
Provide a colour option for table output to stdout

### DISCUSSION THREAD

**Discussion by huornlmj** [aquasecurity/trivy#4363] [Ideas]
As far as I can see, Trivy does not have native HTML report output options, so I went to use the aha tool to pipe the output into, so that I could retain the colouring.  Unfortunately, trivy doesn't seem to have an option to keep colouring when piping the output out.

So when I run:

```console
$ trivy k8s --report summary cluster --timeout 1h | aha > trivy-report-summary.html
```

I end up with an HTML report but just black text, no colour saved.  I would like an option to control whether output colouring is on or off.

**Comment by knqyf263** [comment]
I believe we disable coloring in image/fs scanning when the output is not stdout. 
@josedonizetti @chen-keinan Could you take a look?

**Comment by huornlmj** [comment]
Touching base (avoiding stale state)

**Comment by huornlmj** [comment]
Touching base (avoiding stale state)

**Comment by huornlmj** [comment]
Touching base again.  

Here's an example.  If I want to redirect the colourized tablular output into the aha utility to convert it into an HTML document, trivy is removing the colour as it's being sent to stdout.  There should be a switch so that I can force the colourization so that itr makes it into the aha utility.

```console
$ trivy fs . --colour | aha > trivy-report.txt
```

**Reply by fatihtokus** [reply]
Hi @huornlmj ,

I have a trivy plugin(https://github.com/fatihtokus/scan2html) that creates a well-formated filterable/sortable html report for any scan result of trivy. The usage is very simple, please have a look:

Install the plugin: $ trivy plugin install github.com/fatihtokus/scan2html
$ trivy scan2html k8s --report=all cluster result.html

Regards!
